### PR TITLE
Support for supplying index settings on restore #1362

### PR DESCRIPTION
**Pull request by robertlyson**
PR for https://github.com/elastic/elasticsearch-net/issues/1362 issue.

For `IndexSettings` property on `IRrestoreDescriptor` I reused `IUpdateSettingsRequest`, but I'm not sure if this name fits in this context. Way how we can combine `RestoreDescriptor` with `UpdateSettingsDescriptor` is really nice.

Moreover, I've extracted all setting names to separate class. Avoiding googling for each plain index setting is really convenient.

Any thoughts? Let me know.
Thanks.

### CODE REVIEW

File: src/Nest/DSL/UpdateSettingsDescriptor.cs [hunk 1]
```diff
@@ -1,116 +1,154 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.Linq;
 using Elasticsearch.Net;
 using Newtonsoft.Json;
 
 namespace Nest
-{
+{ 
+	public class SettingNames
```
Comment (Mpdreamz) [inline src/Nest/DSL/UpdateSettingsDescriptor.cs]: Love moving this out of the attributes :+1: suggestion on different name `UpdatableSettings`

**Comment by Mpdreamz** [comment]
PR looks awesome as usual @robertlyson :+1: left a small comment.

File: src/Nest/DSL/UpdateSettingsDescriptor.cs (hunk 1 above)
Comment (robertlyson) [inline src/Nest/DSL/UpdateSettingsDescriptor.cs]: @Mpdreamz much more better name :+1: . Maybe we should split this class into three? UpdatableSettings, UnmodifiableSettings and UnremovableSettings? https://github.com/elastic/elasticsearch/blob/master/src/main/java/org/elasticsearch/snapshots/RestoreService.java#L92-105

Comment (Mpdreamz) [inline src/Nest/DSL/UpdateSettingsDescriptor.cs]: I don't see any settings in `SettingsNames` that are not Updatable though, maybe I need another coffee :smile: 

Where would we use `Unmodifiable` and `Unremovable` ?

Comment (robertlyson) [inline src/Nest/DSL/UpdateSettingsDescriptor.cs]: Yeah, that's true, looks like I need second coffee :)
I'll change name to `UpdatableSettings` later on.

**Comment by robertlyson** [comment]
Thanks for review. I've pushed change.

**Comment by Mpdreamz** [comment]
ty @robertlyson !

**Pull request merged**